### ISSUE TITLE
Wrong vertical scrollbar in mobile filter list

### ISSUE THREAD

**Issue by jordycosta** [iFixit/react-commerce#385]
<details>
<summary> Toggle View </summary> <br>

![scroll](https://user-images.githubusercontent.com/95656772/177365980-bf0d64d2-8382-4791-a354-f07b49fdb669.gif)
</details>

It appears that the filter list may at times use a different vertical scroll bar than what is intended, which can be shown in the above `.gif` file.

**Comment by erinemay** [comment]
> I also found that there are times that it's possible to scroll past the end of the filter list CC @dhmacs

I pulled this into its own issue https://github.com/iFixit/react-commerce/issues/414.

**Comment by sumukhipandey** [comment]
> @jordycosta it's probably scrolling the body underneath. The fix is probably something like locking the body scroll when the drawer is open

@dhmacs I might be misunderstanding  here but doesn't [this line of code](https://github.com/iFixit/react-commerce/blob/7c857922613b241a07ecc542dcb4e6c8296a81c1/frontend/components/product-list/sections/FilterableProductsSection/FacetsDrawer.tsx#L43) already do this?

**Comment by dhmacs** [comment]
> @dhmacs I might be misunderstanding here but doesn't [this line of code](https://github.com/iFixit/react-commerce/blob/7c857922613b241a07ecc542dcb4e6c8296a81c1/frontend/components/product-list/sections/FilterableProductsSection/FacetsDrawer.tsx#L43) already do this?

@sumukhipandey Seems like it 🤔 I know that @federicobadini is working on a related issue on this same part of code, can you coordinate with him on this?

**Comment by sumukhipandey** [comment]
> > @dhmacs I might be misunderstanding here but doesn't [this line of code](https://github.com/iFixit/react-commerce/blob/7c857922613b241a07ecc542dcb4e6c8296a81c1/frontend/components/product-list/sections/FilterableProductsSection/FacetsDrawer.tsx#L43) already do this?
> 
> @sumukhipandey Seems like it 🤔 I know that @federicobadini is working on a related issue on this same part of code, can you coordinate with him on this?

Thank you!

**Comment by federicobadini** [comment]
> I also found that there are times that it's possible to scroll past the end of the filter list CC @dhmacs
> 
>  bypass.mp4

This has been addressed in #435 

> Toggle View
> 
> ![scroll](https://user-images.githubusercontent.com/95656772/177365980-bf0d64d2-8382-4791-a354-f07b49fdb669.gif)
> 
> It appears that the filter list may at times use a different vertical scroll bar than what is intended, which can be shown in the above `.gif` file.

This is related to the fact that iOS Safari has an open [issue](https://bugs.webkit.org/show_bug.cgi?id=153852) regarding handling `overflow: hidden` on the body correctly. The current buggy behaviour is summarized as follows.
```
Scroll on the body is prevented when:
- in Safari in portrait mode when the browser UI is fully visible (i.e. both the address bar and the toolbar bar are visible)
- in Safari in landscape mode (whether the browser UI is fully visible or not)
- in "in-app browsers" (whether the browser UI is fully visible or not)

Scroll on the body is not prevented when:
- in Safari in portrait mode when the browser UI is not fully visible (i.e. the address bar is retracted, and the toolbar bar fully collapsed), whether it is due to a scroll down or with the new "Hide toolbar" feature activated
- in "standalone mode", i.e. when the web page is added to homescreen and no browser UI is displayed.
```

The scenario in the video is one of those in which the scrolling is not prevented and that is the reason why it happens even though `overflow: hidden` is applied.

Using a hard fix like adding `position: fixed` to the body would prevent such behaviour - but will also cause the body to jump to top each time the drawer is opened. Adding some js to modify the body scrollY position after having set it to `position: fixed` is an option too.

But considering that #435 is going to land solving most of the drawer issues, and considering also that in the [follow up issue](https://bugs.webkit.org/show_bug.cgi?id=220908) it seems that a patch for this has landed, I would wait to see if this solves by itself in the following iOS releases.
